intro-HPC/ch_running_batch_jobs.tex

```diff
@@ -40,6 +40,331 @@ \chapter{Running batch jobs}
+
+The toolchains are then used to compile a lot of the software installed on
+the VSC clusters. You can recognise those packages easily as they all contain the
+name of the toolchain after the version number in their name. Packages
+compiled with the same toolchain and toolchain version will typically work
+together well without conflicts.
+
+For some clusters, additional toolchains are defined, e.g., to take advantage
+of specific properties of that cluster such as GPU accelerators or special
+interconnect features that require a vendor-specific MPI-implementation.
+
+\subsection{Activating and de-activating modules}
+\label{subsec:activating-and-deactivating-modules}
+
+A module is loaded using the following command:
+
+
+\begin{prompt}
+%\shellcmd{module load \emph{examplemodule}}%
+\end{prompt}
+
+This will load the most recent version of \emph{examplemodule}.
+
+For some packages (e.g., OpenMPI), multiple versions are installed; the load
+command will automatically choose the most recent version (i.e., the
+lexicographical last after the ``/'') or the default version (as set by the
+system administrators). However, the user can (and probably should, to avoid
+surprises when newer versions are installed) specify a particular version,
+e.g.,
+
+\begin{prompt}
+%\shellcmd{module load \emph{secondexample/2.7-intel-2018a}}%
+\end{prompt}
+
+Obviously, you need to keep track of the modules that are currently
+loaded. If you executed the two load commands stated above, you will get the
+following:
+
+\begin{prompt}
+%\shellcmd{module list}%
+Currently Loaded Modulefiles:
+  1) examplemodule/1.2.3-intel-2018a        3) intel/2018a
+  2) secondexample/2.7-intel-2018a          4) examplelib/1.2-intel-2014b
+\end{prompt}
+
+
+It is important to note at this point that other modules (e.g., \lstinline|intel/2014a|)
+are also listed, although the user did not explicitly load them. This is
+because \lstinline|secondexample/2.7-intel-2018a| depends on it (as indicated in its name),
+and the system administrator specified that the \lstinline|intel/2014a| module should
+be loaded whenever the \lstinline|secondexample| module is loaded. There are advantages and
+disadvantages to this, so be aware of automatically loaded modules whenever
+things go wrong: they may have something to do with it!
+
+In fact, an easy way to check the components and version numbers of those
+components of a toolchain is to simply load the toolchain and then list the
+modules that are loaded.
+
+To unload a module, one can use the \lstinline|module unload| command. It works
+consistently with the load command, and reverses the latter's effect. However,
+the dependencies of the package are NOT automatically unloaded; the user shall
+unload the packages one by one. One can however unload automatically loaded
+modules manually, to debug some problem. When the \lstinline|secondexample| module is unloaded,
```

Comment:
drop the `One can however unload automatically loaded modules manually, to debug some problem.`